bridges/otelzap/zap_logger.go

```diff
@@ -0,0 +1,119 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package otelzap provides a bridge between the ["go.uber.org/zap"] and
+// OpenTelemetry logging.
+package otelzap // import "go.opentelemetry.io/contrib/bridges/otelzap"
+
+import (
+	"context"
+
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+)
+
+const (
+	bridgeName = "go.opentelemetry.io/contrib/bridge/zapcore"
+)
+
+// Core is a [zapcore.Core] that sends all logging records it receives to
+// OpenTelemetry.
+type Core struct {
+	logger log.Logger
+	attr   []log.KeyValue
+}
+
+// Compile-time check *Core implements zapcore.Core.
+var _ zapcore.Core = (*Core)(nil)
+
+// this function creates a new zapcore.Core that can be used with zap.New()
+// this instance will translate zap logs to opentelemetry logs and export them.
+func NewOtelZapCore(opts ...Option) zapcore.Core {
+	cfg := newConfig(opts)
+	return &Core{
+		logger: cfg.logger(),
+	}
+}
+
+// LevelEnabler decides whether a given logging level is enabled when logging a
+// message.
+func (o *Core) Enabled(level zapcore.Level) bool {
+	r := log.Record{}
+	r.SetSeverity(getOtelLevel(level))
+
+	// TODO:
+	// check how to propagate context
```

Comment:
I am not sure if it would be possible.

```suggestion
	// TODO:
	// check how to propagate context
```